src/update.cc

```diff
@@ -874,48 +906,74 @@ void list_patches_by_issue( Zypper & zypper, bool all_r, const PatchSelector &se
 
     for_( it, q.begin(), q.end() )
     {
-      PoolItem pi( *it );
-      Patch::constPtr patch = asKind<Patch>(pi);
+      PoolItem pi { *it };
 
       if ( only_needed && ! patchIsApplicable( pi ) )
 	continue;
 
-      if ( ! cliMatchPatch( patch ) )
+      if ( ! cliMatchPatch( pi ) )
       {
-	DBG << patch->ident() << " skipped. (not matching CLI filter)" << endl;
+	DBG << pi.ident() << " skipped. (not matching CLI filter)" << endl;
 	continue;
       }
 
-      intoSummaryMatchesTbl( pi );
-//	issueMatchesTbl.rows().back().addDetail( str::Str() << "    " << attr.inSolvAttr() << "\t\"" << attr.asString() << "\"" );
-      //! \todo could show a highlighted match with a portion of surrounding
-      //! text. Needs case-insensitive find.
+      if ( ! iresult.count( pi ) )
+      { dresult.push_back( pi ); }
     }
   }
 
+  ///////////////////////////////////////////////////////////////////
   // print result
-  if ( issueMatchesTbl.empty() && summaryMatchesTbl.empty() )
-  {  zypper.out().info(_("No matching issues found.")); }
+  if (zypper.out().type() == Out::TYPE_XML)
+  {
+    xmlout::Node parent { cout, "list-patches-byissue", xmlout::Node::optionalContent };
+    xmlPrintPatchUpdateListOn( *parent, "issue-matches", zypp_pending::make_map_key_Iterable( iresult ) );
+    xmlPrintPatchUpdateListOn( *parent, "description-matches", dresult );
+  }
   else
   {
-    if ( !issueMatchesTbl.empty() )
+    // iresult to table
+    Table issueMatchesTbl;
+    FillPatchesTableForIssue intoIssueMatchesTbl( issueMatchesTbl );
+    for ( const auto & res : iresult )
     {
-      if ( !pass2.empty() )
+      const PoolItem & pi { res.first };
+      for ( const auto & ires : res.second )
       {
-        cout << endl;
-        zypper.out().info(_("The following matches in issue numbers have been found:"));
+	const std::string & itype { ires.first };
+	for ( const std::string & iid : ires.second )
+	{ intoIssueMatchesTbl( pi, itype, iid ); }
       }
-      issueMatchesTbl.sort(); // use default sort
-      cout << endl << issueMatchesTbl;
     }
 
-    if ( !summaryMatchesTbl.empty() )
+    // dresult to table
+    Table descrMatchesTbl;
+    FillPatchesTable intoDescrMatchesTbl( descrMatchesTbl );
+    for ( const auto & pi : dresult )
+    {
+      intoDescrMatchesTbl( pi );
+    }
+
+
+    if ( issueMatchesTbl.empty() && descrMatchesTbl.empty() )
+    {  zypper.out().info(_("No matching issues found.")); }
+    else
     {
       if ( !issueMatchesTbl.empty() )
-      { cout << endl; }
-      zypper.out().info(_( "Matches in patch descriptions of the following patches have been found:"));
-      summaryMatchesTbl.sort(); // use default sort
-      cout << endl << summaryMatchesTbl;
+      {
+	cout << endl;
+	zypper.out().info(_("The following matches in issue numbers have been found:"));
+	issueMatchesTbl.sort(); // use default sort
+	cout << endl << issueMatchesTbl;
```

Comment:
No API for writing a plain Table NORMAL without XML. 

Out::table could be used once the NORMAL and XML codepaths are joined.  It would take a container + formater to render it NORMAL/XML as needed.  This requires further cleanup in the code here (once we have time...). But some day zypper will be cout-free.